includes/snmp.inc.php

```diff
@@ -779,10 +784,10 @@ function snmp_gen_auth(&$device, $cmd = [], $strIndexing = null)
  * @param string $mib
  * @param string $mibdir the mib directory (relative to the LibreNMS mibs directory)
  * @param array|string $options Options to pass to snmptranslate
- * @param array $device
+ * @param array|null $device
  * @return string
  */
-function snmp_translate($oid, $mib = 'ALL', $mibdir = null, $options = null, $device = [])
+function snmp_translate($oid, $mib = 'ALL', $mibdir = null, $options = null, $device = null)
```

Comment:
This should only accept array.

Reply:
No that's what we wanted to get away from.
No point in sending a empty array when we're only interested in it if it's a "device array"

Reply:
Best would ofc be to send the device object instead... One day. One day.